Richardcho/call invite

richardcho-msft opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

m-nash committed:
Regen with property docs 20230201.2 (#33795)

* regen with property docs

* consume latest AD sha

* regen openai

* update autorest version

* update readme

* update to latest sha

* update api and snippets

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 5291 (#33783)

* Move migration script to eng/common

* Remove localization from link

* Ignore the improper link

---------

Co-authored-by: Timo van Veenendaal <timov@microsoft.com>

alanenriqueo committed:
[PostgreSQL] SDK upgrade for PostgreSQL flexible server to version 2022-12-01 (#33602)

azure-sdk committed:
Handle multi-line string concatenation with Write-Warning (#33831)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

joseharriaga committed:
[Text Analytics] Make `FhirBundle` property in `AnalyzeHealthcareEntitiesResult` of type `BinaryData` (#33686)

The gist of this change is simple:
1. Tell AutoRest to leave the internal `FhirBundle` property as a raw `JsonElement`; no need to do work to deserialize it into a `Dictionary` that we won’t use. (see HealthcareEntitiesDocumentResultInternal.cs)
2. Edit the method that transforms the internal result object to its external representation so that the `JsonElement` object is used to create a `BinaryData` object instead. (see Transforms.cs)

schaudhari6254888 committed:
Servicebus Namespace With ScaleSet (#33446)

azure-sdk committed:
Increment version for webpubsub releases (#33837)

Increment package version after release of Microsoft.Azure.WebJobs.Extensions.WebPubSub

archerzz committed:
Revert "[PostgreSQL] SDK upgrade for PostgreSQL flexible server to version 2022-12-01" (#33839)

alanenriqueo committed:
[PostgreSQL] SDK upgrade for PostgreSQL flexible server to version 2022-12-01 (#33843)

Commit:
[Synapse] Add test case for track 2 mgmt sdk - phase 1 (#33801)

azure-sdk committed:
Increment version for postgresql releases (#33847)

Increment package version after release of Azure.ResourceManager.PostgreSql

fengzhou-msft committed:
Support ignoring API versions of some ResourceManager resources when used as test dependencies (#33844)

christothes committed:
Handle SAS credential updates during message retries (#33655)

* Handle SAS credential updates during message retries

sebrenna committed:
[DevCenter] Azure.Developer.DevCenter 1.0.0-beta.2 (#33713)

This pull request includes changes to the Azure.Developer.DevCenter SDK that constitue version 1.0.0-beta.2. This SDK version is built around version **2022-11-11-preview** of our Data Plane API. The most notable change from 1.0.0-beta.1 is the use of an endpoint URI to construct clients, rather than tenant ID + dev center name, which was the architecture board's largest point of feedback for us in our initial review.

jpalvarezl committed:
Addressing AOAI.Inference outstanding comments on documentation (#33827)

* Addressing comment 33691#discussion_r1092413439

* Addressed PR feedback from 33691#discussion_r1092473460

* Removed locale info from link

liliankasem committed:
Blob Storage support for ParameterBindingData reference type (#31757)

* Add converter rules for richbindingreferencetype

* add RichBindingReferenceType rules and converter

* add build script

* update BlobTriggerAttributeBindingProvider

* update ref dictionary

* update local version

* Update to connection name and checking null case

* Renaming to ParameterBindingData

* cleanup

* bind to new ParameterBindingData schema

* update script

* Try content as json string

* use ExpandoObject

* Update Constants file

* Fix input binding

* Use updated ParameterBindingData

* Address feedback

* Update usage of ParameterBindingData and cleanup

* Add azure_app_service_staging and script - both to undo before merge

* address feedback

* Add BlobTrigger_BindingToParameterBindingData test

* add input binding test

* Fix tests

* Validate connection name

* Added input converter for ParameterBindingData

* Removing unnecessary comment

* Updating connection string resolution logic

* Updating unused usings

* Removing duplicate code

* Added test for ParameterBindingData collection

* Removing blank line

* Removing unused constants

* Fixing test

* Adding storage constants

* Added env variable

* Updating blob trigger test

* Adding environment variables

* Add env variables

* Add env variables in tests

* Adding correct using

* Fixing failing test

* Fixing test

* Added createContainer method

* Fix build issue

* Removing comment

* Deleting unused build file

* Removing code duplication to resolve connection

* Resolving connection section

* Removing configuration

* Added nameResolver logic

* Assigning connectionSection.Key instead of value

* Updated test assert

* Updated test assert

* Added connection string in test

* Test fix

* Fixing failing tests

* Fixing failing tests

* Passing configuration in tests for storage setting

* Fix failing tests by sending config correctly

* Test comment

* Updated support for array type

* Updated collection support code

* Adding test for array

* Added test

* Added string array test

* reset NuGet.Config

---------

Co-authored-by: Surgupta <surgupta@microsoft.com>

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 5318 (#33836)

* update target version to newest one that includes .NET body bugfix as well as renames the proxy back to azure.sdk.tools.testproxy

* update package.data.props to align with target_version.txt

---------

Co-authored-by: scbedd <45376673+scbedd@users.noreply.github.com>

azure-sdk committed:
Enable nodepool update via bicep and update node SKUs (#33832)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

liliankasem committed:
Update changelog to include blob support for ParameterBindingData feature (#33863)

Update change log to include new feature supporting ParameterBindingData in the blob storage webjobs ext

PR: https://github.com/Azure/azure-sdk-for-net/pull/31757

heaths committed:
Add docs on running live tests serially (#33829)

* General clean-up of markdownlint issues

* Add section on serial live tests

azure-sdk committed:
Temporarily disable error handling for Get-AzKeyVault -InRemovedState (#33864)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

rajkumar-rangaraj committed:
Azure Monitor Exporter - Fix safe name for Azure.Monitor.OpenTelemetry.Exporter (#33861)

JoshLove-msft committed:
Throw exception if options have already been instrumented (#33828)

Commit:
[AzureMonitorExporter] update Otel packages (#33859)

* update Otel packages

* changelog

jpalvarezl committed:
`deploymentId` moved to be part of `GetCompletions` signature (#33834)

* Removed custom constructors and reinstated deploymentId as part of GetCompletions's signature

* Small corrections to the documentation

* CodeCheck, Export-API, Update-Snippets run

annelo-msft committed:
Prototype moving JsonData to be backed by JsonDocument (#33063)

* Add WriteTo(), ==operators, and PR FB

* export API

* Add DocumentSentiment large JSON sample and add parse and read benchmarks

* initial attempt to move to JsonElement

* some fixes

* updates

* nit

* nits

* starting to experiment with changelist approach

* an approach to object assignment

* start working with array elements

* saving changes from before break...

* sm formatting pr fb

* ApiView FB per generality of DynamicData

* nits

* small thoughts

* first steps toward abstracting ChangeTracker

* In flight changes while implementing AddPropertyToObject

* adding tests with simple modifications to structural elements

* Implementing WriteTo()

* Added nested objects

* refactor

* missed changes

* In TDD spirit, add failing WriteTo test

* Test passes

* quick refactor

* update add property test

* Update WriteTo to handle property additions at the root element

* Handle property additions on arbitrary objects

* handle standard property removals

* Support Replace with object

* Support Replace with object

* refactor where we serialize structural changes to centralize

* Implement reference semantics for JsonDataElement

* experiment with checking ancestors for structural changes.

* Update WriteTo to handle structural changes.

* add high water mark logic

* add validation to all reads and add failing test for ignoring pre-structural change.

* Incorporate HWM logic in all change lookups; make PriorChangeToReplacedPropertyIsIgnored

* remove double-check of object and array elements

* some tidy up

* Reimplement GetProperty in terms of TryGetProperty()

* Handle WriteTo for structural changes.

* add some tests of structural changes

* refactor tests

* Bug fix to WriteTo for bools and test refactoring

* bug fix to WriteTo for booleans

* Add support for nulls and fix ToLower() bug

* add perf benchmark prior to working on perf

* missed file

* update Parse() to use Memory<byte>

* move subclasses to separate files for ease of reading

* refactor to add dynamic layer

* Add cast operators to JsonDataElement to pass dynamic GetIntProperty test

* Add support for dynamic nested property access

* enable set via dynamic layer

* Enable setting nested properties via dynamic layer

* nit

* Renames prior to dynamic refactor

* start adding separate dynamic layer

* Move dynamic meta object to dyanmic types.

* API updates

* todos and nits

* rename test to match types

* remove outer DynamicJson class

* add BindConvert to dynamic layer

* Add BindGetIndex to dynamic layer

* save multitarget attempt

* Add target frameworks to Experimental to allow ifdefs by target framework.

* PR feedback; use more efficient Deserialize call when available in ConvertTo()

* refactor per PR fb

* Add BindSetIndex to dynamic layer

* Fix bug in mutable ToString() where changes to descendants weren't accounted for

* Fix WriteTo() bug for string elements and add failing test for handling nulls

* Handle null values

* PR fb

* Add support for floats and longs

* remove HasValue, per pr fb

* export API changes

* Support adding properties on dynamic member assignments

* reshuffle methods around

* some changes before adding support for IEnumerable

* Add ArrayEnumerator to MutableJsonElement

* missed file; dynamic portion not complete

* Bug fix to dynamic ArrayEnumerator; foreach over array now passes

* export API and misc test updates

* nit; cleanup

* Make MutableJsonDocument serializable

* Make DynamicJson serializable; enable reference semantics for added properties

* Make tests pass for net6.0 & net7.0; will address net461 separately

* Fixes for some net461 issues

* Fix Add and Set property for net461

* Work around BindBinaryOperation in net461; move inline TODOs to GH issue

* Export API

AlexanderSher committed:
Set LangVersion to 11 (#33862)

* Bump LangVersion

* Switch to fixed

JoshLove-msft committed:
Support serializing received message to/from AMQP bytes (#33682)

* Support serializing received message to/from AMQP bytes

* Use ToMemory to avoid extra allocation

* PR fb

* PR fb

* Update test

* Add back TryRead

* Add endianness check

* Fix comment

* Update WebJobs version and remove nuget config change

* Fix build break

Yao725 committed:
[DataProtectionBackup] Polish the API to prepare for release (#33792)

* regen

* regen resources

* resolve comment

* update changelog

azure-sdk committed:
Increment version for dataprotection releases (#33880)

Increment package version after release of Azure.ResourceManager.DataProtectionBackup

archerzz committed:
fix(test): re-enable synapse mgmt sdk tests (#33879)

fix https://github.com/Azure/azure-sdk-for-net/issues/33867

jiriburant committed:
Modified set method response to no longer contain the raw content of the Http response. (#33850)

fengzhou-msft committed:
Support ignoring API versions of more ResourceManager resources when used as test dependencies (#33882)

azure-sdk committed:
Update AutoRest C# version (#33884)

azure-sdk committed:
Increment version for mobilenetwork releases (#33840)

Increment package version after release of Azure.ResourceManager.MobileNetwork

heaths committed:
Fix case-sensitive parameter reference in template (#33868)

See https://github.com/dotnet/templating/issues/5955#issuecomment-1414068854

trrwilson committed:
Azure OpenAI test management update (#33821)

* Initial update

* More progress; token?

* Token-based recordings

* PR feedback and concurrency fix

* Reverting opportunistic changes to client to proactively address merge conflicts with !33834

* PR feedback: concurrency/deployment initialization optimization

* Merged files that weren't merged -- oops!

annelo-msft committed:
Update CHANGELOG for Azure.Core.Experimental release. (#33889)

* prep for release

* Correct release date

annelo-msft committed:
update changelog for 1.1.0-beta.6 release (#33823)

nisha-bhatia committed:
[Azure.Monitor.Ingestion] EventHandler (#33490)

* wip

* wip

* wip

* Update LogsIngestionClient.cs

* wip

* wip

* wip

* wip

* wip

* add tests

* add generic type back to upload methods

* wip

* wip

* wip

* wip

* Update LogsIngestionClient.cs

* Update LogsIngestionClient.cs

* wip

* wip

* wip

* Update LogsIngestionClient.cs

* update data props to test-proxy with additional debug info

* Debug changes

* set to record

* newest proxy version

* Update LogsOptions MaxConcurrency error statement and constructor

* Update Upload method in LogsIngestionClient to create new Options for MaxConcurrency

* new proxy version that includes upstream request dump

* wip

* restore TestProxy

* Update MonitorIngestionLiveTest.cs

* wip

* Update Packages.Data.props

* wip

* Remove cast to List in Upload code

* Update LogsIngestionClient.cs

* Update LogsIngestionClient.cs

* revert snippets

* wip

---------

Co-authored-by: scbedd <45376673+scbedd@users.noreply.github.com>
Co-authored-by: JoshLove-msft <54595583+JoshLove-msft@users.noreply.github.com>

AlexanderSher committed:
Update LangVersion in individual projects (#33890)

* Update LangVersion in individual projects

* Restore LangVersion tag in sample project

* Address PR comment

jpalvarezl committed:
Updated commit and refreshed generated code (#33892)

* Updated commit and refreshed generated code

* Updated commit id

* Removed old namespace from tests

heaths committed:
Update Azure.ClientSdks.Analyzer (#33817)

Support for `NullableResponse<T>` returns

jpalvarezl committed:
Renaming fields for clarity AOAI Inference (#33871)

* Renamed fields added

* Removed empty comment line

* Renamed class file

* update generator and regen

* regen after merge

* update api

* revert unintended version change

---------

Co-authored-by: m-nash <prognash@gmail.com>

JoshLove-msft committed:
Fix Guid handling (#33881)

* Fix Guid handling

* Move to separate class

* using

* Optimize Guid creation

* Optimize big endian write

* Move to core

* Rename

* Fix

* Update parse handling

* Fix SB tests and revert Parse change

* PR fb

* Apply suggestions from code review

Co-authored-by: Daniel Marbach <daniel.marbach@openplace.net>

* Set out param

* Add helper

---------

Co-authored-by: danielmarbach <daniel.marbach@openplace.net>

zackliu committed:
[WebPubSubClient] Prepare for Beta.2 release (#33802)

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

dvbb committed:
Add auto-generate samples for Synapse and Workloads (#33753)

dvbb committed:
Add auto-generate samples for EventGrid and GuestConfiguration (#33750)

dvbb committed:
Add auto-generate samples for Consumption and KubernetesConfiguration (#33748)

dvbb committed:
Add auto-generate samples for DeploymentManager and DnsResolver (#33749)

dvbb committed:
Add auto-generate samples for Blueprint and DataBox (#33747)

dvbb committed:
Add auto-generate samples for RecoveryServicesBackup and SecurityCenter (#33752)

dvbb committed:
Add auto-generate samples for Sql (#33754)

dvbb committed:
Add auto-generate samples for Dns and PrivateDns (#33906)

dvbb committed:
update MarketplaceOrdering and PolicyInsights (#33903)

azure-sdk committed:
Increment package version after release of Azure.Messaging.WebPubSub.Client (#33905)

christothes committed:
Release prep for Core (#33888)

richardcho-msft committed:
add CallInvite

azure-sdk committed:
Increment version for core releases (#33912)

Increment package version after release of Azure.Core

richardcho-msft committed:
add create call options

richardcho-msft committed:
use provided source identity

wangrui-msft committed:
Hide ChoiceResult and Dtmf Result, to use RecognizeResult Only (#33894)

jsquire committed:
[Event Hubs] Allow partition key use in Functions async collector (#33858)

* [Event Hubs] Allow partition key use in async collector

The focus of these changes is to allow for partition keys to be used when
binding to the async collector to publish events.  Previously, use was
restricted to the `IAsyncCollector<T>` interface, which only accepted an
`EventData` instance leaving no way to specify a partition key.  As a result,
all events were assigned partitions via round-robin.

These changes expose the `EventHubAsyncCollector` and allow binding to it,
exposing an `AddAsync` overload specific to Event Hubs which accepts a
partition key.  This allows events to be logically grouped together, ensuring
they are assigned to the same partition.

richardcho-msft committed:
fix linter

richardcho-msft committed:
fix linter

trrwilson committed:
Update CHANGELOG for first Azure.AI.OpenAI preview release (#33918)

richardcho-msft committed:
fix linter

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.Communication.CallAutomation](https://apiview.dev/Assemblies/Review/dc92c96c3c0249c0ac6892f9464cd2bc)

kinelski committed:
[FormRecognizer] Removed unused snippets (#33866)

jsquire committed:
[Event Hubs] Functions extensions doc tweaks (#33923)

The focus of these changes is to tweak some of the documentation for the
Event Hubs function extension around output bindings and `IAsyncCollector<T>`
use.

jsquire committed:
[Smoke Tests] Ignore Known Incompatible Packages (#33925)

* [Smoke Tests] Ignore Known Incompatible Packages

The focus of these changes is to define a set that contains the list of
packages which should not be included as references to the smoke tests.
This is necessary because some compatibility packages do not target
netstandard2.0, and the package source is unable to filter upstream to
restrict to the appropriate packages for the running target.

* Incorporating feedback

richardcho-msft committed:
fix tests

richardcho-msft committed:
Merge branch 'Azure:main' into richardcho/call-invite

heaths commented:
Please rebase on main. This PR touches over 4,000 files and will be impossible to review. Assuming your `upstream` and `origin` remotes are set correctly,

```bash
git fetch upstream main
# on your topic branch...
git rebase upstream/main
# resolve any conflicts
git push --force
```

Thank you.

richardcho-msft committed:
reset

richardcho-msft committed:
Merge branch 'richardcho/call-invite' of https://github.com/richardcho-msft/azure-sdk-for-net into richardcho/call-invite
